server/migrations/20210719151134-make-matchmaking-preferences-based-on-type.js

```diff
@@ -0,0 +1,77 @@
+const sql = require('sql-template-strings')
+
+exports.up = async function (db) {
+  await db.runSql(`ALTER TABLE matchmaking_preferences ADD COLUMN data jsonb;`)
+  // This column is not used anymore so it's not necessary
+  await db.runSql(`ALTER TABLE matchmaking_preferences DROP COLUMN updated_at;`)
+
+  // This should probably be batched, maybe? Shouldn't be an issue with our current amount of users,
+  // I think.
+  const { rows: preferences } = await db.runSql(`
+    SELECT user_id, matchmaking_type, use_alternate_race, alternate_race
+    FROM matchmaking_preferences;
+  `)
+
+  await Promise.all(
+    preferences.map(p => {
+      const data = {
+        useAlternateRace: p.use_alternate_race,
+        alternateRace: p.alternate_race,
+      }
+
+      return db.runSql(sql`
+        UPDATE matchmaking_preferences
+        SET data = ${data}
+        WHERE user_id = ${p.user_id} AND matchmaking_type = ${p.matchmaking_type};
+      `)
+    }),
+  )
+
+  await db.runSql(`
+    ALTER TABLE matchmaking_preferences
+    RENAME COLUMN preferred_maps
+    TO map_selections;
+  `)
+  await db.runSql(`ALTER TABLE matchmaking_preferences ALTER COLUMN map_selections SET NOT NULL;`)
```

Comment:
This was nullable before and this migration does not update any data in the table to ensure there are no nulls, so, you should probably be doing that first.